### PR TITLE
Don't allow primitives to be null, as well as reference types

### PR DESCRIPTION
**Pull request by Aaron1011**
See issue https://github.com/SpongePowered/SpongeAPI/issues/439 for a description of the problem.

This **correctly** breaks the build - many event factories don't specif all the needed parameters to construct an event. The fixes can be added to this PR, or go in a separate one.

### CODE REVIEW

**Comment by Minecrell** [comment]
We need to add the fixes to the cause, so it should be fixed in this PR (although preferably as separate commit)

File: src/main/java/org/spongepowered/api/util/event/factory/ClassGenerator.java [hunk 1]
```diff
@@ -81,6 +84,7 @@
 
     private final PropertySearchStrategy propertySearch = new AccessorFirstStrategy();
     private NullPolicy nullPolicy = NullPolicy.DISABLE_PRECONDITIONS;
+    private final List<String> primitivePropertyExceptions = Collections.singletonList("cancelled");
```
Comment (Minecrell) [inline src/main/java/org/spongepowered/api/util/event/factory/ClassGenerator.java]: Do you use the List here so new exceptions can be added easily? If so, it should probably use `ImmutableList.of` instead so you don't need to switch the collection type and can add new elements directly to the method call

File: src/main/java/org/spongepowered/api/event/entity/living/player/PlayerDeathEvent.java [hunk 2]
```diff
@@ -53,7 +53,7 @@
      *
      * @return Whether the player keeps their inventory on death
      */
-    boolean keepsInventory();
+    boolean getKeepsInventory();
```
Comment (Minecrell) [inline src/main/java/org/spongepowered/api/event/entity/living/player/PlayerDeathEvent.java]: Is this necessary? Is there no way to generate the event class without the prefix?

Comment (gratimax) [inline src/main/java/org/spongepowered/api/event/entity/living/player/PlayerDeathEvent.java]: This was discussed in IRC. No.

**Comment by modwizcode** [comment]
@Aaron1011 This builds correctly at least API wise, what was broken?

**Comment by Aaron1011** [comment]
@modwizcode: I've fixed it, per [@Minecrell's comment](https://github.com/SpongePowered/SpongeAPI/pull/471#issuecomment-76529220). Some event factories didn't take all the necessary primitive parameters (such as `PlayerDeathEvent`). This caused the build to be broken, since missing primitives counted as a failure.

**Pull request closed without merging**